### PR TITLE
allow to override cluster controller manager for CI

### PR DESCRIPTION
**Pull request by aojea**
/kind cleanup
```release-note
NONE
```

This allows to override the CCM image or to specify a custom manifest, the same way that it is done for other components.

This is based on the CUSTOM_INGRESS logic https://github.com/kubernetes/test-infra/pull/11747/files , that allows to run periodic jobs with the master of the ingress-gce project against the master of kubernetes

### CODE REVIEW

**Comment by aojea** [comment]
/assign @BenTheElder 
for approval

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -2323,6 +2322,13 @@ function start-cloud-controller-manager {
 
   echo "Applying over-rides for manifest for cloud provider controller-manager"
   local -r src_file="${KUBE_HOME}/kube-manifests/kubernetes/gci-trusty/cloud-controller-manager.manifest"
+
+  if [ -n "${CUSTOM_CLOUD_CONTROLLER_MANAGER_YAML:-}" ]; then
+      # Replace with custom cloud controller manager.
+      cat > "$src_file" <<EOF
+$CUSTOM_CLOUD_CONTROLLER_MANAGER_YAML
+EOF
+  fi
```
Comment (aojea) [inline cluster/gce/gci/configure-helper.sh:2331]: this was missing, that is why it was failing

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

**Comment by liggitt** [comment]
I don't think we should keep extending the cluster directory to allow injecting images to test external projects. If we're serious about ratcheting this directory down to nothing and removing it, projects that want to test against Kubernetes should set up their own clusters off of CI builds, and overlay their own images, not rely on these cluster scripts

**Comment by aojea** [comment]
> I don't think we should keep extending the cluster directory to allow injecting images to test external projects

it makes sense, I was doing it because my understanding is that the in-tree cloud provider will be removed and the CI will have to run always with the out-of-tree cloud provider and I thought this was the easier way to test it.

> projects that want to test against Kubernetes should define jobs that set up their own clusters off of CI or PR builds, and overlay their own images, not rely on these cluster scripts

that works for me too

**Comment by andrewsykim** [comment]
> allow injecting images to test external projects.

To clarify, supporting this is not for testing external projects (cloud-provider-gcp), but it's to support kubernetes/kubernetes CI when we remove in-tree providers.  

>  projects that want to test against Kubernetes should define jobs that set up their own clusters off of CI or PR builds, and overlay their own images, not rely on these cluster scripts

I agree, and cloud-provider-gcp already has it's own version of `cluster/` for testing, but this is for testing changes specific to cloud-provider-gcp, not in k/k.

**Pull request closed without merging**